libcxx/include/__atomic/atomic_sync.h

```diff
@@ -70,42 +77,97 @@ __libcpp_atomic_monitor(__cxx_atomic_contention_t const volatile*) _NOEXCEPT;
 _LIBCPP_EXPORTED_FROM_ABI void
 __libcpp_atomic_wait(__cxx_atomic_contention_t const volatile*, __cxx_contention_t) _NOEXCEPT;
 
+// new dylib interface
+_LIBCPP_AVAILABILITY_NEW_SYNC _LIBCPP_EXPORTED_FROM_ABI __cxx_contention_t
+__libcpp_atomic_monitor_global(void const volatile* __address) _NOEXCEPT;
```

Comment:
Is there a reason for using `volatile` pointers all over the place? If there's a good reason, then fine, but let's document it in a comment. If not, let's remove it.

Reply:
I added `volatile` in the new API simply because the old API also has `volatile`. Thinking it a bit more, perhaps the reason why we had `volatile` in the old API perhaps because it wanted to allow the inputs to be a `volatile` object?

Reply:
Actually, the OS API takes a straight void pointer, not even const or volatile. So at the end of the day, we do end up casting away all of this fancyness.

I would suggest making the dylib API take non-volatile pointers and then casting away the volatile in the headers, since the volatile is pointless anyway. Let's not hardcode that in our ABI just to satisfy the type system.

Of course, this is predicated on the assumption that `volatile` support in `atomic::wait` & friends was added for consistency with `atomic`'s C++11 APIs and serves no real purpose. If we find otherwise, the above doesn't hold anymore.